core/quantity/src/Formatter/Formatter.ts

```diff
@@ -414,10 +437,104 @@ export class Formatter {
     else
       formattedValue = formattedMagnitude;
 
-    if (spec.format.minWidth && spec.format.minWidth < formattedValue.length)
-      formattedValue.padStart(spec.format.minWidth, " ");
-
     return formattedValue;
   }
 
+  private static processBearingAndAzimuth(magnitude: number, spec: FormatterSpec): {magnitude: number, prefix?: string, suffix?: string} {
+    const type = spec.format.type;
+    if (type !== FormatType.Bearing && type !== FormatType.Azimuth)
+      return {magnitude};
+
+    const revolution = this.getRevolution(spec);
+    magnitude = this.normalizeAngle(magnitude, revolution);
+    const quarterRevolution = revolution / 4;
+
+    if (type === FormatType.Bearing) {
+      let quadrant = 0;
+      while (magnitude > quarterRevolution) {
+        magnitude -= quarterRevolution;
+        quadrant++;
+      }
+      let prefix, suffix: string;
+
+      // Quadrants are
+      // 1 0
+      // 2 3
+      // For quadrants 0 and 2 we have to subtract the angle from quarterRevolution degrees because they go clockwise
+      if (quadrant === 0 || quadrant === 2)
+        magnitude = quarterRevolution - magnitude;
+
+      // TODO: at some point we will want to open this for localization, in the first release it's going to be hard coded
+      if (quadrant === 0 || quadrant === 1)
+        prefix = "N";
+
+      if (quadrant === 2 || quadrant === 3)
+        prefix = "S";
+
+      if (quadrant === 0 || quadrant === 3)
+        suffix = "E";
+
+      if (quadrant === 1 || quadrant === 2)
+        suffix = "W";
+
+      // special case, if in quadrant 2 and value is very small, turn suffix to E because S00E is preferred over S00W
+      if (quadrant === 2 && magnitude < 0.00000000001) // 1e-11 is a small value that is close to 0
```

Comment:
I'm kind of surprised that this is the first place where we've had to use some small value to represent 'almost zero'.

In this case I'm not sure what you've done is enough.  If you never want to show S00W then shouldn't you be comparing the formatted value rather than the raw magnitude?  The precision of the formatted output might treat much larger values as zero

Reply:
I'll have to think about that for a bit. Imagine the unit is whole degrees, and the value is 0.4. Which formats to "0" can/should we consider that zero? I feel like I'm not qualified to make that decision. We do have an AlmostEqual() method in native quantity.

Reply:
Well you don't make that decision the format's definition makes the decision how many decimal digits to show.  That's why I was suggesting that you have to look at it after formatting the double value instead of looking at the double value.

Reply:
For bearing, we usually use more than one presentation unit in a composite.
That formatting happens after this code.
So what I had to do to achieve the logic you describe was I had to convert the unit and compare it using the format's precision. The check is merely a preference, both results would be fine, but Kevin said "E" is preferred.
The adjusted code is a bit gross, but the alternative would be breaking encapsulation of bearing inside this method, and putting some bearing logic outside after the formatting code, which looked even more gross.
Please have a look at the updated code.